hash/hash.go

```diff
@@ -0,0 +1,101 @@
+// Package hash provides utility functions to process complex data types, like
+// data streams, files, or sequences of structs of different types.
+package hash
+
+import (
+	"bytes"
+	"errors"
+	h "hash"
+	"io"
+	"os"
+
+	"encoding"
+	"reflect"
+)
+
+var defaultChunkSize = 1024
+
+// Stream returns the hash of a data stream separated into chunks of the defaultChunkSize.
+func Stream(hash h.Hash, stream io.Reader) ([]byte, error) {
```

Comment:
What is the difference with that function and the `File`-function? Can't they be merged?